charts/cluster-api-runtime-extensions-nutanix/templates/helm-repository.yaml

```diff
@@ -26,6 +26,21 @@ spec:
     kind: Issuer
   secretName: helm-repository-tls
 ---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  finalizers:
+  - kubernetes.io/pvc-protection
```

Comment:
The clusterctl upgrade process deletes provider components and redeploys them. But since we are not providing label to the PVC, it should not delete it. https://github.com/kubernetes-sigs/cluster-api/blob/main/cmd/clusterctl/client/cluster/components.go#L143

It would be good to test out upgrade capi-components process.

Since default retainPolicy for our `default` storage class is `delete` it will delete the PV if PVC is deleted. we have to ensure this PVC is not deleted automatically by any of our processes.

Reply:
yes, im taking a look at that now.

Reply:
the labels are automatically applied it would be up to the client to remove them. we did this downstream by implementing a componentsClient t